chips/earlgrey/src/chip.rs

```diff
@@ -110,6 +114,11 @@ impl<'a, A: 'static + Alarm<'static>, I: InterruptService<()> + 'a> EarlGrey<'a,
 
     unsafe fn handle_plic_interrupts(&self) {
         while let Some(interrupt) = self.plic.get_saved_interrupts() {
+            self.atomic(|| {
+                // Safe as interrupts are disabled
+                self.plic.complete(interrupt);
+            });
+
```

Comment:
Yep, I have added this in the code as well:

```rust
            // Claim the interrupt before we handle it.
            // Currently the interrupt has been claimed but not completed.
            // This means that if the interrupt re-asserts we will loose the
            // re-assertion. Generally this isn't a problem, but some of the
            // interrupt handlers expect that interrupts could occur.
            // For example the HMAC interrupt handler will write data to the
            // HMAC buffer. We then rely on an interrupt triggering when that
            // buffer comes empty. This can happen while we are still in the
            // interrupt handler. To ensure we don't loose the interrupt we
            // claim it here.
            // In order to stop an interrupt loop, we first disable the
            // interrupt.
```